pkg/terraform/gather/aws/ip.go

```diff
@@ -19,11 +19,14 @@ func BootstrapIP(tfs *terraform.State) (string, error) {
 	if len(br.Instances) == 0 {
 		return "", errors.New("no bootstrap instance found")
 	}
-	bootstrap, _, err := unstructured.NestedString(br.Instances[0].Attributes, "public_ip")
-	if err != nil {
-		return "", errors.New("no public_ip found for bootstrap")
+
+	for _, att := range []string{"public_ip", "private_ip"} {
+		ip, _, _ := unstructured.NestedString(br.Instances[0].Attributes, att)
+		if ip != "" {
+			return ip, nil
+		}
 	}
-	return bootstrap, nil
+	return "", errors.New("no usable IP found for bootstrap instance")
```

Comment:
Do we want to throw the `NestedString` errors out and create a new one here?  How about:

```go
var err error
for _, att := range []string{"public_ip", "private_ip"} {
  var ip string
  ip, _, err = unstructured.NestedString(br.Instances[0].Attributes, att)
  if err != nil {
    return ip, nil
  }
}
return "", err
```

Or ending with `return "", errors.Wrap(err, "looking up bootstrap IP")` or some such.

Reply:
the error would add no more information, it will always be set to the private_ip get failed, if it ever happens.. And I think `no usable IP found` is more accurate in that case.

Reply:
/hold cancel